0pdd.rb

```diff
@@ -213,11 +213,24 @@
 
 get '/snapshot' do
   content_type 'text/xml'
-  repo = repo(repo_name(params[:name]))
-  repo.push
-  xml = repo.xml
-  xml.xpath('//processing-instruction("xml-stylesheet")').remove
-  xml.to_s
+  name = repo_name(params[:name])
+  master = params[:branch]
+  uri = "git@github.com:#{name}.git"
+  begin
+    repo = GitRepo.new(
+      uri: uri,
+      name: name,
+      id_rsa: settings.config['id_rsa'],
+      dir: settings.temp_dir,
+      master: master || 'master'
+    )
+    repo.push
+    xml = repo.xml
+    xml.xpath('//processing-instruction("xml-stylesheet")').remove
+    xml.to_s
+  rescue Exec::Error => e
```

Comment:
@mbao01 can you please explain WHY do you need special error handling in this particular web entry point? why not in all others?